src/engine/sparqlExpressions/NumericBinaryExpressions.cpp

```diff
@@ -68,9 +66,198 @@ inline auto andLambda = [](TernaryBool a, TernaryBool b) {
+    std::optional<std::vector<PrefilterExprVariablePair>>&& rightChildExprs) {
+  if (!leftChildExprs.has_value() || !rightChildExprs.has_value()) {
+    // If one of the children yields no PrefilterExpressions, we simply can't
+    // further define a PrefilterExpression with this OR (||) conjunction.
+    //
+    // EXAMPLE / REASON:
+    // Assume that left is - {(=10, ?x), (=10, ?y)}
+    // and right is - std::nullopt ({undefinable prefilter})
+    // Remark: left represents the expression ?x = 10 && ?y = 10.
+    // We can't define a PrefilterExpression here because left must not always
+    // yield true when this OR expression evaluates to true (because right can
+    // plausibly yield true).
+    // And because we can't definitely say that one child (and which of the two
+    // children) must be true, pre-filtering w.r.t. underlying compressed blocks
+    // is not possible here.
+    // In short: OR has multiple configurations that yield true w.r.t. two child
+    // expressions => we can't define a distinct PrefilterExpression.
+    return std::nullopt;
+  }
+  auto& expressionsLeft = leftChildExprs.value();
+  auto& expressionsRight = rightChildExprs.value();
+  auto itLeft = expressionsLeft.begin();
+  auto itRight = expressionsRight.begin();
+  std::vector<PrefilterExprVariablePair> resPairs;
+  while (itLeft != expressionsLeft.end() && itRight != expressionsRight.end()) {
+    auto& [exprLeft, varLeft] = *itLeft;
+    auto& [exprRight, varRight] = *itRight;
+    // The logic that is implemented with this loop.
+    //
+    // EXAMPLE 1
+    // left child: {(>=5, ?y)}
+    // right child: {(<=3, ?x)}
+    // We want to pre-filter the blocks for the expression: ?y >= 5 || ?x <= 3
+    // No PrefilterExpression will be returned.
+    //
+    // EXAMPLE 2
+    // left child: {(>=5, ?x)}
+    // right child: {(=0, ?x)}
+    // We want to pre-filter the blocks to the expression: ?x >= 5 || ?x = 0
+    // The resulting PrefilterExpression is {((>=5 OR =0), ?x)}
+    //
+    // EXAMPLE 3
+    // left child: {(>=10, ?x), (!=0, ?y)}
+    // right child: {(<= 0, ?x)}
+    // We have to construct a PrefilterExpression for (?x >= 10 && ?y != 0) ||
+    // ?x <= 0. If this OR expression yields true, at least ?x >= 10 || ?x <= 0
+    // must be staisifed; for this objective we can construct a
+    // PrefiterExpression. We can't make a distinct prediction w.r.t. ?y != 0
+    // => not relevant for the PrefilterExpression. Thus, we return the
+    // PrefilterExpresion {((>= 10 OR <= 0), ?x)}.
+    if (varLeft == varRight) {
+      resPairs.emplace_back(std::make_unique<BinaryPrefilterExpr>(
+                                std::move(exprLeft), std::move(exprRight)),
+                            varLeft);
+      ++itLeft;
+      ++itRight;
+    } else if (varLeft < varRight) {
+      ++itLeft;
+    } else {
+      ++itRight;
+    }
+  }
+  if (resPairs.empty()) {
+    return std::nullopt;
+  }
+  return resPairs;
+}
+
```

Comment:
(empty) comment missing.

Reply:
It seems that these only appear in the `.cpp` file and not in the header
(also the above functions). Then they should be
* documented (short docstring)
* made `static` (either by marking them  static or by putting them into the anonymous `namespace { ...}` (which can also be combined with other namespaces.